types/styled-components/test/index.tsx

```diff
@@ -1049,10 +1049,23 @@ function unionTest() {
     };
 
     const StyledReadable = styled(Readable)`
-        font-size: ${props => props.kind === 'book' ? 16 : 14}
+        font-size: ${props => props.kind === 'book' ? 16 : 14};
     `;
 
     // undesired, fix was reverted because of https://github.com/Microsoft/TypeScript/issues/30663
     <StyledReadable kind="book" author="Hejlsberg" />; // $ExpectError
     <StyledReadable kind="magazine" author="Hejlsberg" />; // $ExpectError
 }
+
+/* forwardedAs test */
+function forwardedAsTest() {
+    const ChildComponent = () => <div>foo</div>;
```

Comment:
The child should take an `as` prop as well, no? Otherwise `forwardedAs` would do nothing.